pkg/policyviolation/builder.go

```diff
@@ -3,55 +3,46 @@ package policyviolation
 import (
 	"fmt"
 
+	"github.com/golang/glog"
 	kyverno "github.com/nirmata/kyverno/pkg/api/kyverno/v1"
-	client "github.com/nirmata/kyverno/pkg/dclient"
+	"github.com/nirmata/kyverno/pkg/engine/response"
 )
 
+func GeneratePVsFromEngineResponse(ers []response.EngineResponse) (pvInfos []Info) {
+	for _, er := range ers {
+		// ignore creation of PV for resoruces that are yet to be assigned a name
+		if er.PolicyResponse.Resource.Name == "" {
+			glog.V(4).Infof("resource %v, has not been assigned a name, not creating a policy violation for it", er.PolicyResponse.Resource)
+			continue
+		}
+		// skip when response succeed AND referenced paths exist
+		if er.IsSuccesful() && !er.IsPathNotPresent() {
```

Comment:
In reference to some above comments,  the pathNotPresent is engineResponse error instead of adding a new filed and check specifically for it.
As this is one of the multiple check-in policy, others use a common field rule failure, which for this field we add a new field and new error checks.

Reply:
Again, as we gonna report pathNotPresent as the policy error, it would be helpful to decouple this type of error from violation.